### DISCUSSION TITLE
Allow "on" keyword in  RSpec/ContextWording

### DISCUSSION THREAD

**Discussion by D4uS1** [rubocop-hq/rubocop-rspec#1270] [Ideas]
Hey guys, 

in my opinion it would be nice to allow the "on" keyword for contexts in the  RSpec/ContextWording cop.
This would be especially nice to test rails hooks that works "on update" or "on create" or "on destroy".

But i guess this is some matter of oppinion, hence i started this discussion.

What do you think about this?

**Comment by pirj** [comment]
Hey!

I'll separate in topics.

### Configurable

The [Community RSpec Style Guide is quite flexible in this regard](https://github.com/rubocop/rspec-style-guide#context-descriptions):
> A typical description will be an adjunct phrase starting with 'when', 'with', 'without', or similar words

The [`rubocop-rspec`'s default config](https://github.com/rubocop/rubocop-rspec/blob/af2fd8a216b0ed7846372672a9eadf5db87126b2/config/default.yml#L195) has those three:
```yaml
  Prefixes:
    - when
    - with
    - without
```
but you are encouraged to reconfigure that in your project or company-wide RuboCop settings to meet your style.


### Popularity

A [3 y.o. research](https://github.com/rubocop/rubocop-rspec/pull/757#issuecomment-492448802) indicated the following frecency of prefix use:
- when 23760
- with 8150
- for 2090
- and 1993
- as 1204
- without 1035
- in 719
- user 688
- **on 509**
- if 482

Things might have changed since then. But still, "on" was far from the top of the list.

### `context` vs `describe`

It's subjective, and is my personal opinion.
There is a subtle difference between semantics of `context` and `describe`, even though they are technically just [aliases of `example_group`](https://github.com/rspec/rspec-core/blob/bc9bb2da423b2cf6b7e53f1307c3ef2d24fceb7b/lib/rspec/core/example_group.rb#L280).
`context` is generally used when grouping examples contextually (e.g. "with xyz", "when xyz" or "if xyz"). I may have understood the meaning of "on update"/"on create" mistakenly, but it appears to me that those example groups consist of examples that are grouped not by the context (similar setup), but by what they test. In such a case, a `describe` might be a better way of grouping them. Please correct me if I'm mistaken with this hypothesis.
